records/2024-04-05.database-activity-logging.md

```diff
@@ -0,0 +1,74 @@
+# Database activity logging
+
+Date: 2024-04-05
+
+## Status
+
+**DRAFT**
+
+## Context
+
+To improve GC Notify's security posture, database activity logging should be enabled.  This will give the product and security teams visibility into the database activity.  It will also allow for the creation of automated alerting rules to detect malicious or unusual behaviour.  
+
+This work is related to the most recent Security Assessment and Authorization (SA&A) performed on Notify, which identified the [AU-6 - Audit Review, Analysis, and Reporting](https://www.cyber.gc.ca/en/guidance/annex-3a-security-control-catalogue-itsg-33#a33au6) control as required.
+
+## Options
+
+There are two options for auditing database activity with an RDS Aurora PostgreSQL database cluster:
+
+1. [database activity streams](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/DBActivityStreams.Overview.html); and
+2. the [PostgreSQL audit extension (pgAudit)](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Appendix.PostgreSQL.CommonDBATasks.pgaudit.html).
+
+### Option 1: database activity streams
+
+A database activity stream logs all queries performed by the databases that are part of an Aurora cluster. The activity stream is encrypted with a KMS Customer Managed Key (CMK) and can be integrated with other AWS services.  No database instance configuration or storage space is needed to setup and use the activity stream.
+
+The proposed solution for GC Notify would be to create an asynchronous activity stream to minimize database performance impact.  This stream would then be ingested by a Kinesis Firehose which would:
+
+1. decrypt the activity events,
+1. filter events that should not be captured; and 
+1. write activity logs to an S3 bucket for auditing and alerting.
```

Comment:
Are the logs stored in S3 encrypted as well?

Reply:
Yes, they would be encrypted at rest with the S3 service key, similar to how other S3 objects in Notify are stored.  It does mean anyone with access to the S3 bucket would be able to read those query logs.